kernel/src/grant.rs

```diff
@@ -165,15 +165,29 @@ impl<T: Default> Grant<T> {
             appid
                 .kernel
                 .process_map_or(Err(Error::NoSuchApp), appid.idx(), |process| {
-                    process.grant_for_or_alloc::<T>(self.grant_num).map_or(
-                        Err(Error::OutOfMemory),
-                        move |root_ptr| {
-                            let mut root = Borrowed::new(&mut *root_ptr, appid);
-                            let mut allocator = Allocator { appid: appid };
-                            let res = fun(&mut root, &mut allocator);
-                            Ok(res)
-                        },
-                    )
+                    let ctr_ptr = process.grant_ptr(self.grant_num) as *mut *mut T;
```

Comment:
Could you add a comment to the top of this block that clarifies that you're either getting a pointer to the grant, or else allocating it.